orangecontrib/text/pubmed.py

```diff
@@ -0,0 +1,480 @@
+        self.search_record_count = 0  # Number of records.
+        self.search_record_web_env = None  # BioPython history param.
+        self.search_record_query_key = None  # BioPython history param.
+
+        self.progress_callback = progress_callback
+        self.error_callback = error_callback
+
+        self.cache_path = None
+        cache_folder = os.path.join(environ.buffer_dir, 'pubmedcache')
+        try:
+            if not os.path.exists(cache_folder):
+                os.makedirs(cache_folder)
+            self.cache_path = os.path.join(cache_folder, 'query_cache')
+        except:
+            warnings.warn('Could not assemble Pubmed query cache path',
+                          RuntimeWarning)
+
+    def _search_for_records(self, terms=[], authors=[],
+                            pub_date_start=None, pub_date_end=None):
+        """Executes a search for PubMed records.
+
+        Results of the search, like QueryKey, WebEnv and Count are stored as
+        parameters of this class's instance.
+
+        Args:
+            terms (list): Keywords to search by.
+            authors (list): Authors to search by.
+            pub_date_start (str): The start of the publish date chronological
+                interval.
+            pub_date_end (str): The end of the publish date chronological
+                interval.
+        """
+        search_terms = []
+
+        argument_checks = [
+            {
+                'argument': authors,
+                'error': '{} is not a valid data type for the field "authors".'
+                    .format(type(authors)),
+                'conjunction': ' AND ',
+                'field': 'Author',
+            },
+            {
+                'argument': terms,
+                'error': '{} is not a valid data type for the field "terms".'
+                    .format(type(terms)),
+                'conjunction': ' OR ',
+                'field': 'All fields',
+            },
+        ]
+
+        for check in argument_checks:
+            if not isinstance(check.get('argument'), list):
+                raise ValueError(check.get('error'))
+            arguments_list = ['{}[{}]'.format(arg, check.get('field'))
+                              for arg in check.get('argument')]
+            arguments = check.get('conjunction').join(arguments_list)
+            if arguments:
+                search_terms.append('({})'.format(arguments))
+
+        # The 'retmax' parameter determines the maximum number of record ids
+        # we want returned, where 100000 is the maximum. Setting it to maximum
+        # will therefore return all available record ids.
+        try:
+            search_handle = Entrez.esearch(
+                    db='pubmed',
+                    term=' AND '.join(search_terms),
+                    mindate=pub_date_start,
+                    maxdate=pub_date_end,
+                    datetype='pdat',
+                    usehistory='y',
+                    retmax='100000'
```

Comment:
Does this has to set to 100,000? If yes, could you store it in a variable like `DEFAULT_BATCH_SIZE` so it is only set in one place?